apps/api/v2/src/modules/oauth/flow/oauth-flow.controller.ts

```diff
@@ -0,0 +1,84 @@
+import { GetUser } from "@/modules/auth/decorator";
+import { NextAuthGuard } from "@/modules/auth/guard";
+import { OAuthAuthorizeInput } from "@/modules/oauth/flow/input/authorize.input";
+import { ExchangeAuthorizationCodeInput } from "@/modules/oauth/flow/input/exchange-code.input";
+import { OAuthClientGuard } from "@/modules/oauth/guard/oauth-client/oauth-client.guard";
+import { OAuthClientRepository } from "@/modules/oauth/oauth-client.repository";
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Headers,
+  HttpCode,
+  HttpStatus,
+  Logger,
+  Post,
+  UseGuards,
+} from "@nestjs/common";
+
+import { SUCCESS_STATUS, X_CAL_CLIENT_ID } from "@calcom/platform-constants";
+import { ApiResponse } from "@calcom/platform-types";
+
+@Controller({
+  path: "oauth",
+  version: "2",
+})
+export class OAuthFlowController {
+  private logger = new Logger("OauthFlowController");
+
+  constructor(private readonly oauthClientRepository: OAuthClientRepository) {}
+
+  @Post("/authorize")
+  @HttpCode(HttpStatus.OK)
+  @UseGuards(OAuthClientGuard, NextAuthGuard)
```

Comment:
this endpoint is called from the front-end cal.com app, I think the oAuthGard is not needed, frontend can send client_id in the body